src/platform_specific/named_threads.win.cc

```diff
@@ -0,0 +1,69 @@
+#include "named_threads.h"
+
+#include <cassert>
+#include <cwchar>
+#include <type_traits>
+
+#include <windows.h>
+
+namespace celerity::detail {
+
+static_assert(std::is_same_v<std::thread::native_handle_type, HANDLE>, "Unexpected native thread handle type");
+
+template <typename SrcT, typename DstT>
+static inline DstT convert_string(const SrcT& str) {
+	static_assert(
+	    (std::is_same_v<SrcT, std::string> && std::is_same_v<DstT, std::wstring>) || (std::is_same_v<SrcT, std::wstring> && std::is_same_v<DstT, std::string>),
+	    "Unsupported string type");
+
+	constexpr auto converter = [](typename DstT::value_type* dst, const typename SrcT::value_type** src, const std::size_t len, std::mbstate_t* ps) {
+		if constexpr(std::is_same_v<SrcT, std::string>) {
+			return std::mbsrtowcs(dst, src, len, ps);
+		} else {
+			return std::wcsrtombs(dst, src, len, ps);
+		}
+	};
+
+	const auto* src = str.c_str();
+	auto mbstate = std::mbstate_t{};
+	const auto len = converter(nullptr, &src, 0, &mbstate);
+	auto dst = DstT(len, L'\0'); // Automatically includes space for the null terminator
+	converter(dst.data(), &src, dst.size(), &mbstate);
+	return dst;
+}
+
+static inline void set_thread_name_windows(const HANDLE thread_handle, const std::string& name) {
+	const auto wname = convert_string<std::string, std::wstring>(name);
+	[[maybe_unused]] const auto res = SetThreadDescription(thread_handle, wname.c_str());
+	assert(SUCCEEDED(res) && "Failed to set thread name");
+}
+
+static inline std::string get_thread_name_windows(const HANDLE thread_handle) {
+	PWSTR wname = nullptr;
+	const auto res = GetThreadDescription(thread_handle, &wname);
+	assert(SUCCEEDED(res) && "Failed to get thread name");
+	std::string name;
+	if(SUCCEEDED(res)) {
+		name = convert_string<std::wstring, std::string>(wname);
+		LocalFree(wname);
```

Comment:
`wname` will leak when `convert_string` throws. This might not be a real concern as this method is only used in tests, but maybe add a comment.

Reply:
The only exception that `convert_string` could throw is `bad_alloc` (unless I'm missing something), and if that happens the situation is FUBAR anyway. I briefly thought about catching/rethrowing to properly free `wname`, but I don't think it matters in this case, so I just added the comment.